File: bindings/python/examples/getting_started.py
"""getting-started: one-shot JSONLogic evaluation with datalogic-py,
plus the typed-result tier for boolean predicates.

Run from bindings/python/ (build first: maturin develop --release):
    python examples/getting_started.py
"""

from datalogic_py import DataHandle, Engine, apply

rule = {
    "and": [
        {">=": [{"var": "age"}, 18]},
        {"==": [{"var": "status"}, "active"]},
    ]
}
data = {"age": 25, "status": "active"}

# One-shot: compile + evaluate in a single call.
print(apply(rule, data))  # True

# Typed result: for predicates, skip the JSON round trip entirely.
# Compile the rule, parse the data once into a handle, and read the
# result directly as a Python bool.
engine = Engine()
compiled = engine.compile(rule)
parsed = DataHandle('{"age": 25, "status": "active"}')

session = engine.session()
print(session.evaluate_bool(compiled, parsed))  # True

File: bindings/node/examples/getting-started.mjs
// getting-started: one-shot JSONLogic evaluation, plus the typed-result
// tier for boolean predicates and the async string tier.
//
// Run from bindings/node/ (build first: npm install && npx napi build --platform --release):
//   node examples/getting-started.mjs

import { apply, Engine, DataHandle } from '../index.js';

const rule = {
  and: [
    { '>=': [{ var: 'age' }, 18] },
    { '==': [{ var: 'status' }, 'active'] },
  ],
};
const data = { age: 25, status: 'active' };

// One-shot: compile + evaluate in a single call.
console.log(apply(rule, data)); // true

// Typed result: for predicates, skip the JSON result round trip.
// Compile the rule, parse the data once into a handle, and read the
// result directly as a boolean.
const engine = new Engine();
const compiled = engine.compile(rule);
const parsed = new DataHandle(JSON.stringify(data));

const session = engine.session();
console.log(session.evaluateBool(compiled, parsed)); // true

// Async: evaluate on the libuv pool — same result, off the event loop.
console.log(await compiled.evaluateStrAsync(JSON.stringify(data))); // true
